### ISSUE TITLE
Test and fix any issues with Nested Collections in the app and docs

### ISSUE THREAD

**Issue by KaneFreeman** [StaticJsCMS/static-cms#63]
Nested Collection support is currently a Beta feature from NetlifyCMS. In its current state it remains untested after the rewrite and may be potentially broken and not working at all.

- [x] Test out feature
- [x] Fix any issues found
- [x] Update documentation as necessary

Docs: http://localhost:3000/docs/collection-types#nested-collections

**Comment by Skrubbadubba** [comment]
I have some tests to share:
![image](https://user-images.githubusercontent.com/75796723/220340256-3a631762-42d1-4677-a422-1f61c345e8fa.png)

That is how my nested-collection currently looks. "student" is the only entry inside "sidor" (swedish for pages) that has its own entries, so the dropdown and list of entries are viewing what they should. Obviously the collection isn't showing correctly (the decorated anchor tag and smaller font size).

Icons do not work.

When clicking on an entry, it shows an empty background. There is no error message here unfortunately, neither on the page or in console.

When clicking on a nested-collection from the dropdown, we get an error. Clickin on "sidor" in my case directs to "http://localhost:3000/admin/#/collections/sidor/filter/student" and shows:
![image](https://user-images.githubusercontent.com/75796723/220342156-3c852799-6bfe-4390-b710-8c7d61a46231.png)

Doesn't say much, but the console thankfully says where the error actually occured.
`Collection.tsx:90:18` which has this code:
```tsx
  const newEntryUrl = useMemo(() => {
    let url = 'fields' in collection && collection.create ? getNewEntryUrl(collectionName) : '';
    if (url && filterTerm) {
      url = getNewEntryUrl(collectionName);
      if (filterTerm) {
        url = `${newEntryUrl}?path=${filterTerm}`; //<-- this is row 90
      }
    }
```

**Comment by superboum** [comment]
I encounter the same problem as @Skrubbadubba

**Comment by lyind** [comment]
In addition to the routing issues for nested entries addressed in #677, I encountered a potential performance issue:

All nested objects (documents) are loaded when visiting `/#`, lazy loading would be in order to handler sites with a bigger number of documents.

**Comment by KaneFreeman** [comment]
This is resolved in the v2.0.0 branch (`next`)
